### PR TITLE
If someone else is grabbing entity you want to grab show their grab beam

### PR DESCRIPTION
**Pull request by ctrlaltdavid**

### CODE REVIEW

**Comment by ctrlaltdavid** [comment]
Testing

If you try to distance grab an entity but someone else is grabbing it:
- You laser beam stays red (i.e., doesn't change to the magenta "grabbing" colour).
- A laser beam is drawn to the entity from the avatar that is grabbing it.
If the other person stops grabbing while you're still trying to grab the entity, you successfully start grabbing it.

**Comment by hifi-gustavo** [comment]
Build console output for commit: d9a716bf3de4bc95ddd0d98f5ecd0e5a3487cc10 available at: https://gist.github.com/1e4fc1dd6099c7fa5b9d94d6b4e2a5c7<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9749-d9a716b.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9749-d9a716b.dmg

**Comment by ctrlaltdavid** [comment]
Test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: d9a716bf3de4bc95ddd0d98f5ecd0e5a3487cc10 available at: https://gist.github.com/1e4fc1dd6099c7fa5b9d94d6b4e2a5c7<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9749-d9a716b.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9749-d9a716b.dmg

File: scripts/system/controllers/handControllerGrab.js [hunk 1]
```diff
@@ -419,10 +419,10 @@ function entityIsGrabbedByOther(entityID) {
         }
         if (tag.slice(0, 5) == "grab-") {
             // we see a grab-*uuid* shaped tag and it's not ours, so someone else is grabbing it.
-            return true;
+            return tag.slice(5, 42);
```
Comment (ZappoMan) [inline scripts/system/controllers/handControllerGrab.js]: can you ad some comment about why 5/42 is correct... maybe actually use some named variables for the 5 and 42 presumably 5 is `GRAB_PREFIX_LENGTH` and 42 is maybe `GRABBER_ID_LENGTH`

**Comment by hifi-gustavo** [comment]
Build console output for commit: a98824f48300179684c5e3817545cbf67de1bfc6 available at: https://gist.github.com/1e4fc1dd6099c7fa5b9d94d6b4e2a5c7<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9749-a98824f.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9749-a98824f.dmg

**Comment by nbq** [comment]
Testing this with @Nex-Pro

**Comment by Nex-Pro** [comment]
Testing this with @fossilizedOrganicMatter

**Comment by Nex-Pro** [comment]
I meant @nbq

**Comment by Nex-Pro** [comment]
If you try to distance grab an entity but someone else is grabbing it:

You laser beam stays red (i.e., doesn't change to the magenta "grabbing" colour). **SUCCESS**
A laser beam is drawn to the entity from the avatar that is grabbing it.**FAIL**
If the other person stops grabbing while you're still trying to grab the entity, you successfully start grabbing it. **FAIL**

@nbq and I both saw magenta beams from the avatar trying to take control but the beams eminated from the hips and did not connect with the Entity that was being manipulated.

**Comment by ctrlaltdavid** [comment]
The other avatar's grab beam is  now drawn from a position up from and in front of the avatar's hips (instead of a position in front of the avatar's hips) and goes to the grabbed entity's centroid (instead of the entity's registration point.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 6fcc096bcf57cd2d3429906e3ceec3e16dd5e140 available at: https://gist.github.com/1e4fc1dd6099c7fa5b9d94d6b4e2a5c7<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9749-6fcc096.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9749-6fcc096.dmg

**Comment by PhilipRosedale** [comment]
Looks good now to me.   Basically what we have done here is disallow two people far grabbing at once, with a very minimal indication.  Which is better than fighting over something you can't see being grabbed.  
I don't think it is the right long-term solution, as it makes grabbing something a negative experience - "I can't grab this" rather than a positive one - "I can see that you are grabbing this and I will politely wait".

**Pull request merged**